### DISCUSSION TITLE
Is deno ecosystem production ready now?

### DISCUSSION THREAD

**Discussion by loynoir** [denoland/deno#15101] [General]
Although I am big fan of `deno`'s future, which can 
- use typescript out of box 
- without `nodejs`'s annoying not chainable loader, or wait compiling, leading to many configurations

I am still hesitating to switch from `node` to `deno` now.

Emm, here are some questions...
- ~~Is `deno` **ecosystem** production ready now?~~
> This depends on what your requirements are for production software
- ~~Which companies and apps are using `deno`?~~
https://github.com/denoland/deno/wiki#companies-interested-in-deno
- Is there a workflow, self hostable, one can publish fully typescript package, and use it in browser, and also in node for compactibility?
- ~~How does registries guarantee to future-reproducible build?~~ 
https://deno.land/manual/linking_to_external_code/integrity_checking
- Is there a `git`-ish registry like [cdnjs/cdnjs](https://github.com/cdnjs/cdnjs)?
- Is there a cheat sheet, cover famous `nodejs` library's deno equivalents, or high lighted no equivalent, like koa to oak, listed in style like [rosetta](https://wiki.archlinux.org/title/Pacman/Rosetta)?
- Is there a Hot-Module-Replacement guide?
- ~~Is there a way to alter partial dependency code, without effect other projects?~~
I think there is a `deno vendor` download dependencies to local dir, which can path library locally instead of global modification.

Thanks.

### Additional
I think it would be nice to have some kind of deno server standard
- show dependents in `deno info X`
- how many times N were package X is downloaded from central in recent Y time
- how many times N were package X is referenced from cache in recent Y time

**Comment by zachauten** [comment]
This probably belongs in the [discussions](https://github.com/denoland/deno/discussions) section, as it's a question an not really a bug report / issue. That said i'll do my best to try to answer!
> Is deno ecosystem production ready now?

This depends on what your requirements are for production software, but Deno's been stable since v1.0, and there are plenty of [3rd party modules](https://deno.land/x) for things like db drivers, web frameworks, etc.

> Is there a git-ish registry like [cdnjs/cdnjs](https://github.com/cdnjs/cdnjs)?

Not sure what you mean by "git-ish", but Deno supports ES module imports, so you can host it on a cdn and import it via the url.

> Is there a cheat sheet, cover famous nodejs library's deno equivalents, or high lighted no equivalent, like koa to oak, listed in style like [rosetta](https://wiki.archlinux.org/title/Pacman/Rosetta)?

I don't think the Deno team has created anything like this, but if you search the third party deps, discussions & discord server, you could probably easily figure out if there's a Deno equivalent for any given node package. There's also a [compatibility layer](https://deno.land/manual/node) for running node projects in Deno.

> Is there a Hot-Module-Replacement guide?

There's `deno run --watch`, but I don't think there's hot module replacement. Looks like there's a tracking issue (#8327) for it though.

> Is there a way to alter partial dependency code, without effect other projects?

Not sure what you mean by this, sorry.

Hope this was helpful!

**Reply by sant123** [reply]
Here is a Deno cheat sheet 😄 https://oscarotero.com/deno/

**Reply by jespertheend** [reply]
Here's another cheat sheet https://twitter.com/lcasdev/status/1483507347859378182

**Reply by sant123** [reply]
That is based on https://deno.land/manual/node/cheatsheet 😁

**Reply by jespertheend** [reply]
Ah I see xD
